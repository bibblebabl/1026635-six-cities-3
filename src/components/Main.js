import React from 'react';
import PropTypes from 'prop-types';

const Main = ({data}) => {
  return (
    <div>
      <ul>
        {data.map((el) =>
          <li key={el.id}>
            <span>{el.country}</span> // <span>{el.city}</span>
          </li>
        )}
      </ul>
    </div>
  );
};

Main.propTypes = {
  data: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    city: PropTypes.string,
    country: PropTypes.string
  }))
};

export default Main;
