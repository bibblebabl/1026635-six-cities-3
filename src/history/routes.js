const Routes = {
  MAIN: `/`,
  SIGN_IN: `/login`,
  FAVORITES: `/favorites`,
  OFFER: `/offer`
};

export default Routes;
