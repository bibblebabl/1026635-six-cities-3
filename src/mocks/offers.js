export default [
  {
    id: 1,
    city: `Amsterdam`,
    title: `Beautiful & luxurious studio at great location`,
    image: `img/apartment-01.jpg`,
    price: 120,
    rating: 4.8,
    type: `apartment`,
    isFavorite: false,
    isPremium: false,
  },
  {
    id: 2,
    city: `Amsterdam`,
    title: `Beautiful & luxurious studio at great location`,
    image: `img/apartment-02.jpg`,
    price: 120,
    rating: 4.8,
    type: `apartment`,
    isFavorite: true,
    isPremium: false,
  },
  {
    id: 3,
    city: `Amsterdam`,
    title: `Beautiful & luxurious studio at great location`,
    image: `img/apartment-03.jpg`,
    price: 120,
    rating: 4.8,
    type: `apartment`,
    isFavorite: false,
    isPremium: false,
  },
  {
    id: 4,
    city: `Amsterdam`,
    title: `Beautiful & luxurious studio at great location`,
    image: `img/apartment-01.jpg`,
    price: 120,
    rating: 4.8,
    type: `apartment`,
    isFavorite: false,
    isPremium: false,
  },
  {
    id: 5,
    city: `Amsterdam`,
    title: `Beautiful & luxurious studio at great location`,
    image: `img/apartment-02.jpg`,
    price: 120,
    rating: 4.8,
    type: `apartment`,
    isFavorite: false,
    isPremium: false,
  },
];
