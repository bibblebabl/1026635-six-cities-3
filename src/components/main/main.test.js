import React from 'react';
import renderer from 'react-test-renderer';

import Main from './main';

import offers from '../../mocks/tests/offers';
import {Router} from 'react-router-dom';
import history from '../../history/history';

it(`<Main /> renders correctly`, () => {

  const props = {
    offers,
    cities: [`Amsterdam`, `Hamburg`, `Brussels`, `Cologne`],
    selectedCity: {
      "name": `Amsterdam`,
      "location": {
        "x": 52.37454,
        "y": 4.897976,
        "zoom": 13
      }
    },
    user: null,
    sortingType: `Popular`,
    handlePlaceCardMouseOver: jest.fn(),
    handleTitleClick: jest.fn(),
    handleCityNameClick: jest.fn(),
    handleChangeSortingType: jest.fn(),
  };


  const component = renderer.create(
      <Router history={history}>
        <Main
          {...props}
        />
      </Router>,
      {
        createNodeMock: () => document.createElement(`div`)
      })
  .toJSON();

  expect(component).toMatchSnapshot();
});
