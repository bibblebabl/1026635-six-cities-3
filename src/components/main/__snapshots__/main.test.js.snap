// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> renders correctly 1`] = `
<div
  className="offers"
>
  <h1
    className="offers__title"
  >
    Предложения аренды
  </h1>
  <ul>
    <li>
      <span>
        China
      </span>
       // 
      <span>
        Lushuihe
      </span>
    </li>
    <li>
      <span>
        Thailand
      </span>
       // 
      <span>
        Nong Ruea
      </span>
    </li>
  </ul>
</div>
`;
